query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -228,6 +228,53 @@ class QuantileOverTimeFunction(funcParams: Seq[Any]) extends RangeFunction {
   }
 }
 
+class MedianAbsoluteDeviationOverTimeFunction(funcParams: Seq[Any]) extends RangeFunction {
+  override def addedToWindow(row: TransientRow, window: Window): Unit = {
+  }
+
+  override def removedFromWindow(row: TransientRow, window: Window): Unit = {
+  }
+
+  override def apply(
+                      startTimestamp: Long, endTimestamp: Long, window: Window,
+                      sampleToEmit: TransientRow,
+                      queryConfig: QueryConfig
+                    ): Unit = {
+    val q = 0.5
+    val values: Buffer[Double] = Buffer.ofSize(window.size)
+    var i = 0
+    while (i < window.size) {
+      val curValue = window.apply(i).getDouble(1)
+      if (!curValue.isNaN) {
+        values.append(curValue)
+      }
+      i = i + 1
+    }
+    val counter = values.length
+    values.sort(spire.algebra.Order.fromOrdering[Double])
+    val (weight, upperIndex, lowerIndex) = QuantileOverTimeFunction.calculateRank(q, counter)
+    var median : Double = Double.NaN
+    if (counter > 0) {
+      median = values(lowerIndex)*(1-weight) + values(upperIndex)*weight
+    }
+
+    var medianAbsoluteDeviationResult : Double = Double.NaN
+    val diffFromMedians: Buffer[Double] = Buffer.ofSize(window.size)
+
+    while (i < window.size) {
+      val curValue = window.apply(i).getDouble(1)
+      diffFromMedians.append(Math.abs(median - curValue))
+      i = i + 1
+    }
```

Comment:
Can this be removed? `i` will always be `== window.size` here, I think.
Actually, how is `diffFromMedians` populated?